File: services/billing-service/Models/Invoice.cs
namespace BillingService.Models;

public sealed class Invoice
{
    public Guid Id { get; set; }
    public int Number { get; set; }
    public InvoiceStatus Status { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? ClosedAtUtc { get; set; }
    public List<InvoiceItem> Items { get; set; } = [];
}

File: frontend/src/app/core/models/invoice.ts
export interface InvoiceItem {
  productId: string;
  productCode: string;
  productDescription: string;
  quantity: number;
}

export interface Invoice {
  id: string;
  number: number;
  status: 'Open' | 'Closed';
  createdAtUtc: string;
  closedAtUtc: string | null;
  items: InvoiceItem[];
}

export interface CreateInvoiceItemRequest {
  productId: string;
  quantity: number;
}

export interface CreateInvoiceRequest {
  items: CreateInvoiceItemRequest[];
}
